# Results

We demonstrate ... / summarize results or something here


![
**Organisms used in this paper**
](images/tree_fig.png "tree fig"){#fig:tree_fig}

## Reduced alphabet k-mers accurately estimate microbial pangenomes

To determine whether pangenomes could be constructed using amino acid or other reduced alphabet k-mers, we constructed pangenomes from amino acid (k = 7, 8, 9, 10, 11), dayhoff (k = 13, 15, 17), and the hydrophobic-polar (k = 27, 31) encodings and compared these against pangenomes constructed from genes using roary. 
We constructed pangenomes for 23 species from 23 phyla in the GTDB taxonomy, ranging from 20 to 972 genomes per pangenome (average = 203).
For each pangenome, we compared the total number of genes and k-mers and the total number of unique genes and k-mers for each genome. 
We also tested whether genomes that had similar gene presence-absence profiles had similar k-mer presence-absence profiles using the Mantel test.
Performance varied minimally across encodings (**Figure @fig:violin_fig**) thus we performed the majority of our analyses using amino acid encodings with a k-mer size of 10.

+ @Tessa: Do we need to justify selecting a k-mer size of 10 here?
+ @Tessa: Do I need to show a comparison to a scaled of 1 here? I have it for 12 of the species, and the numbers are similar enough to scaled 100 that I stopped wasting the compute on scaled 1
+ @Tessa: should I compare this against nucleotide k-mers at all? Bc I think that's the underlying assumption here, nucleotides don't work for this stuff.

![
**K-size and alphabet do not impact pangenome estimation with k-mers.** Violin plots represent the distrubtion of R2 values for linear models (Total, Unique) or statistic values for mantel tests (Mantel). *Total* corresponds to correlation between total number of distinct genes and distinct k-mers in a genome. *Unique* corresponds to correlation between the number of unique genes and unique k-mers in genome. *Mantel* corresponds to a mantel test between the gene and k-mer presence-absence matrices.
](images/violin_fig.png){#fig:violin_fig}

Performance across metrics varied dramatically for different pangenomes, with k-mers and genes highly correlated for some pangenomes and not correlated for others. 
We investigated pangenomes more closely to determine the source of the poor correlations and found that they were caused by the presence of many frameshifted proteins, one of many potential criteria for exclusion of GenBank genomes from RefSeq.
For example, *Leptospira interrogans* had an R^2^ of 0.12, however 21 genomes contained frameshifted proteins. 
Removing these genomes increased the R^2^ to 0.87 (**Figure @fig:frameshift_fig A,B**).
This trend was consistent across pangenomes, where pangenomes that had at least one genome that was excluded from RefSeq for having many frameshifted proteins had a significantly lower R^2^ values between total number of genes and total number of k-mers per genome than pangenomes that did not (Welch Two Sample t-test, estimate = -0.36, p = 0.003) (**Figure @fig:frameshift_fig C**).

![
**Genomes that are excluded from RefSeq for having many frameshifted proteins reduce similarity between gene- and k-mer-based pangenomes.**
**A, B)** Scatterplots of the total number of distinct genes and k-mers per genome for the species *Leptospira interrogans*, where each point represents a single genome in the pangenome. 
Removing genomes flagged with RefSeq exclusion criteria "many frameshifted proteins" improves the correlation between these variables. 
**C)** Boxplot of R^2^ values between the total number of distinct genes and k-mers per genome. 
Pangenomes that contain genomes with the RefSeq exclusion criteria of 'many frameshifted proteins' have significantly lower R^2^ values
](images/frameshift_fig.png){#fig:frameshift_fig}

We next investigated whether other pangenome metrics were well correlated between k-mer-based and gene-based methods using pangenomes that did not contain genomes excluded from RefSeq for many frameshifted proteins.
For these 13 pangenomes, the percent of k-mers or genes predicted to be part of the core, shell, or cloud genome was strongly correlated (**Figure @fig:pg_fig**).
We also compared whether pangenomes would be designated as open or closed by calculating the alpha value for the Heaps law model. 
Alpha was also strongly correlated between gene- and k-mer based pangenomes (**Figure @fig:pg_fig**). 

![
**Pangenome metrics strongly correlate between gene- and k-mer-based pangenomes.** Pangenome categories core, shell, and cloud refer to genes or k-mers shared between the majority (>95%), some, or singleton genomes in the pangenome. \Alpha is a value from Heaps law used to estimate whether a pangenome is open or closed.
](images/pg_fig.png){#fig:pg_fig}

Lastly, we investigated whether the relationship between the number of k-mers in coding domain sequences in a genome and the number of genes in a genome was constant across diverse phyla represented here. (**Figure @fig:all_pw_fig**)

+ @Tessa: Is this surprising? I know we discussed this on slack a little, but I guess with the way pangenomes are constructed, maybe its not surprising?

![
**The ratio of total distinct genes per genome to total distinct k-mers per genome is conserved across distantly related species.** 
Each point represents a single genome, and genomes are colored by species.
](images/all_pw_fig.png){#fig:all_pw_fig}


## Jaccard similarity between reduced alphabet k-mers and k-mers in database accurately predicts open reading frames for short sequencing data

Given that protein k-mers can accurately estimate bacterial and archaeal pangenomes, we next sought to determine whether open reading frames could be accurately predicted directly from short sequencing reads, as this would enable pangenome analysis without assembly.
We evaluated whether orpheum, a tool recently developed to predict open reading frames in Eukaryotic short reads (CITE), could also perform this this task in bacterial and archaeal sequences.
Orpheum predicts open reading frames by comparing reduced alphabet k-mers in six frame translations of short sequencing reads against those in a database (jaccard similarity) and assigns an open reading frame as coding if similarity exceeds a user-defined threshold (CITE).
To evaluate orpheum, we constructed a database from all k-mers from coding domain sequences from genomes in GTDB rs202.
Using representative genomes from the 23 species above, as well as 20 additional genomes not in the GTDB rs202 database, we simulated short sequencing reads either from coding domain sequences or non-coding sequences in the genome and used these simulated reads to test orpheum.

Orpheum accurately separated coding from non-coding reads for reads that were simulated from genomes in GTDB (**Figure @fig:orpheum_fig A**).

+ Why FN/FP?
  + pseudogenes? fragments? sequencing error?
  
For reads simulated from genomes not in GTDB, orpheum recovered the majority of coding reads when genomes of the same species were in the database (**Figure @fig:orpheum_fig A,B**).

![
**Orpheum correctly assigned short sequencing reads as coding or non-coding and selects the correct open reading frame.**
**A)** Percent of simulated coding or non-coding sequences predicted as coding, non-coding, or discarded based on quality metrics (see methods). 
Genomes are split by those in GTDB and those not in GTDB. 
**B)** Boxplots of the percent of coding reads that were recovered by Orpheum, separated by the level of taxonomic assignment acheived by GTDB-Tk. 
Orpheum recovers more coding sequences when there are closely related genomes in the database. 
**C)** Databases constructed of only closely-related genomes recover the majority of coding sequences, but including increasingly distantly related genomes improves total coding recall. 
**D)** Barchart of the jaccard similarity cutoff that maximized the number of true positives for coding and non-coding reads. 
The best jaccard similarity decreases when there are fewer closely related genomes in the database. 
**E)** XXX.
](images/orpheum_fig.png){#fig:orpheum_fig}

**OUTLINE**

+ CDS/nonCDS, in GTDB/not in GTDB results
+ what are FP/FN
  + pseudogenes? fragments? "sequencing error"?
+ decreasing accuracy with fewer closely related genomes in database
+ both genomes of the same species and genomes from distantly related organisms are important for ORF recovery
  + horizontal gene transfer
+ lower jaccard similarity to maximize true positives with increasingly distantly related genomes
  + but TP decrease at lower jaccard similarity for genomes that have many genomes of the same species in the DB. Pseudogenes?
+ k-mer size, alphabet selection
  + we don't have any evidence for this yet with these data sets.
+ Even with FP/FN rate, comp matrix + mantel test to show we get the same biological results.
+ Should/do I have to compare these results against FragGeneScan?

## K-mer-based metapangenomics combined with assembly graphs ... 

+ Do I need to do this? It's sort of the point, and while both methods are usable on their own for different workflows, it doesn't all work together without the assembly graph component
+ What's the simplest possible way to do this?
  + Do it for two samples, Hu SB1 and SB2? Could pick a species that has bins in both samples, and that we queried with for the sgc paper.
  + Do it with a time series iHMP person? I have a snakefile that gets half the way there already (https://github.com/taylorreiter/2021-sgc-arg).
  + scoop one of the 54 nbhds I already have for all 605 IBD samples? or some subset of them? I'd really rather not get into that whole schtick, and I don't have MAGs for those samples (although I guess pasolli or something might...)
+ Do I need to compare these results against typical metapangenomics? like do de novo assembly, binning, prokka? etc?

+ If I do this -- show that k-mers support pangenome visualization by demonstrating. If not, incorporate somewhere else.

